cpp/ql/src/semmle/code/cpp/ir/implementation/aliased_ssa/Instruction.qll

```diff
@@ -297,7 +297,8 @@ class Instruction extends Construction::TStageInstruction {
   /**
    * Gets the opcode that specifies the operation performed by this instruction.
    */
-  final Opcode getOpcode() { result = Construction::getInstructionOpcode(this) }
+  pragma[inline]
+  final Opcode getOpcode() { Construction::getInstructionOpcode(result, this) }
```

Comment:
Is `pragma[inline]` actually needed here?

Reply:
Yes. Without it, `Instruction::getOpcode` is compiled with the wrong column order, and we get yet another SCAN that does a reordering of the columns:
```
Tuple counts for Instruction::Instruction::getOpcode_dispred#ff_10#join_rhs/2@1f13c0:
  15657421 ~3%     {2} r1 = SCAN Instruction::Instruction::getOpcode_dispred#ff OUTPUT In.1 'arg0', In.0 'arg1'
                    return r1
...
Tuple counts for Instruction::CallInstruction#class#f/1@df6792:
  586586   ~0%     {1} r1 = JOIN Instruction::Instruction::getOpcode_dispred#ff_10#join_rhs WITH construct<TOpcode,51>@dom#Opcode::TCall#0#f ON FIRST 1 OUTPUT Lhs.1 'this'
                    return r1
```
and with the `inline` we get this clean piece of RA:
```
Tuple counts for Instruction::CallInstruction#class#f/1@3c2d81:
  586586   ~0%     {1} r1 = JOIN construct<TOpcode,51>@dom#Opcode::TCall#0#f WITH IRConstruction::getInstructionOpcode#ff ON FIRST 1 OUTPUT Rhs.1 'this'
                    return r1
```